### PR TITLE
To do app with React and Lifecycle methods

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Added React class components with props and state and later replaced with hooks
- Added event handling and raising of state
- Implemented CSS using inline styling & CSS modules
- Implemented persistence between component state and localStorage
- Implemented lifecycle methods componentDidMount, componentDidUpdate & componentWillUnmount and later replaced with useEffect
- Implemented React Hooks and function components
- Implemented React Icons from font awesome.
- Implemented React Router
- Added a hamburger menu

### CODE REVIEW

File: src/components/InputTodo.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (inputText.title.trim()) {
+      addTodoProps(inputText.title);
+      setInputText({
+        title: '',
+      });
+    } else {
+      alert('Please write item');
+    }
```
Comment (didierganthier) [inline src/components/InputTodo.js]: In order to follow [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) you should avoid using `window.alert` which is the same as `alert` considering the global state of the `window` object

File: src/components/TodoItem.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  const editMode = {};
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  useEffect(
+    () => () => {
+      console.log('Cleaning up...');
```
Comment (didierganthier) [inline src/components/TodoItem.js]: In order to follow [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) you should avoid committing `console.log` on the repo, it kills performance.

**Review by didierganthier — APPROVED** [review]
Hi @JonahKayizzi,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [x] The Pull Request has a proper title and description. ✔️ 
- [x] Descriptive `README.md` file. ✔️ 
- [x] No linters error. ✔️ 
- [x] The project is working as expected ✔️ 


## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/TodoItem.js (hunk 2 above)
Comment (JonahKayizzi) [reply src/components/TodoItem.js]: Thanks for the recommendation. I will implement it and merge

**Pull request merged**